src/stp/new_plays/Attack.cpp

```diff
@@ -3,65 +3,116 @@
 //
 
 #include "stp/invariants/WeHaveBallInvariant.h"
+#include "stp/invariants/GoalVisionInvariant.h"
+#include "stp/invariants/BallCloseToUsInvariant.h"
+#include "stp/invariants/BallIsFreeInvariant.h"
+#include "stp/invariants/game_states/NormalPlayGameStateInvariant.h"
 #include "stp/new_plays/Attack.h"
 #include "stp/new_roles/Attacker.h"
-#include "stp/new_roles/TestRole.h"
+#include "stp/new_roles/Keeper.h"
+#include "stp/new_roles/Formation.h"
+#include "stp/new_roles/Defender.h"
 
 namespace rtt::ai::stp::play {
 
 Attack::Attack() : Play() {
-    // TODO: decide start invariants
     startPlayInvariants.clear();
+    startPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
     startPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveBallInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::GoalVisionInvariant>());
 
-    // TODO: decide keep invariants
-/*    keepPlayInvariants.clear();
-    keepPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveBallInvariant>());*/
+    keepPlayInvariants.clear();
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallCloseToUsInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
```

Comment:
So this play will only be kept if we don't have the ball? How can we shoot the ball then?

Reply:
Oops hehe